app/code/Magento/Catalog/Model/Product/Type/Price.php

```diff
@@ -368,10 +368,33 @@ protected function getAllCustomerGroupsId()
      * @return \Magento\Catalog\Api\Data\ProductTierPriceInterface[]
      */
     public function getTierPrices($product)
+    {
+        $tierPricesRaw = $this->getExistingPrices($product, 'tier_price');
+
+        return $this->buildProductTierPriceInterfaceObjects($tierPricesRaw);
+    }
+
+    /**
+     * Return ProductTierPriceInterface[] given raw tier prices array
+     *
+     * @param array $tierPricesRaw
+     * @return \Magento\Catalog\Api\Data\ProductTierPriceInterface[]
+     */
+    public function transformTierPrices($tierPricesRaw)
```

Comment:
Adding a public method to an "@api" class is a backwards incompatible change, that is best avoided.

Reply:
I tink the alternative is to extract \Magento\Catalog\Model\Product\Type\Price::buildProductTierPriceInterfaceObjects and \Magento\Catalog\Model\Product\Type\Price::getWebsiteForPriceScope into another class..
we can deprecate getWebsiteForPriceScope and buildProductTierPriceInterfaceObjects

Later in 2.5.0 we can delete them.
in 2.4.1 in theory we can't even add this new class to the constructor with "= null" Backward compatible way unless we have an approval. we can try.
Anyway either that or code duplication.

Reply:
For this purpose, created now is Magento\Catalog\Model\Product\Price\TierPriceBuilder.